### DISCUSSION TITLE
how to set `publishDiagnostic/tagSupport` for `pyright`

### DISCUSSION THREAD

**Discussion by yoomaxxx** [helix-editor/helix#8444] [Q&A] [answered]
how to set `publishDiagnostic/tagSupport` capability for `pyright`, like neovim 
https://github.com/neovim/nvim-lspconfig/issues/726#issuecomment-1439132189
https://github.com/neovim/nvim-lspconfig/issues/726#issuecomment-1439257063

**Comment by pascalkuthe** [comment]
helix does not support diagnostic tags and does not indicate as such in its capabilities so this doesn't apply to us.

We could enable support for them in the future but then there would be no way to disable it (although if we do support them we would only support them properly by rendering striketrough/greyed out text). I don't think we want to allow user to customize tags in their confugation. It is not possible to update capabilities without restarting the server and LSP capabilities are really not meant for user configuration (the actual configuration can be updated with `DidChangeConfiguration`)

**Comment by yoomaxxx** [comment]
> helix does not support diagnostic tags and does not indicate as such in its capabilities so this doesn't apply to us.
> 
> We could enable support for them in the future but then there would be no way to disable it (although if we do support them we would only support them properly by rendering striketrough/greyed out text). I don't think we want to allow user to customize tags in their confugation. It is not possible to update capabilities without restarting the server and LSP capabilities are really not meant for user configuration (the actual configuration can be updated with `DidChangeConfiguration`)

Better not add with *no way to disable it*, I wanted to turn them off, like in neovim, but it turns out there is simply no support

**Reply by pascalkuthe** [reply, marked as answer]
I think this is an issue with the python lsp. If they have diagnostics that some users may not want to see than they should provide an option to turn them off. Event if we displayed them the exact same way vscode does they would still be visible just less so.

Treating capabilities as a user config is not a good idea. Just because a clinet is able to properly display theses tags doesn't necessarily imply the user wants to a see a specific diagnostic. It would be less intrusive if the variable was only gray but I guess some users would not even want that (and we would likely still the show the message if the cursor is placed there).

Other language servers allow all diagnostics (including such hits) to be gradually enabled and disabled. I don't see why pyright can't do that. Blocking support for a standard compliant implementation of an LSP feature on some people not liking how some LSP servers implement this feature (and providing no option to configure it) would be strange.
